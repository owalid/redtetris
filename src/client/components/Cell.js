import React from 'react';
import StyledCell from './style/styledCell'
import { TETROMINOS } from '../plugins/tetrominos'

const Cell = ({ type, isOtherUser }) => 
  <StyledCell
    type={type}
    color={(!isOtherUser || type === 0) ? TETROMINOS[type].color : '40,40,40'}
  />

export default Cell
