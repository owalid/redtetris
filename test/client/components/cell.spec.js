import React from "react";
import '@testing-library/jest-dom/extend-expect';
import Enzyme, { shallow } from "enzyme";
import { render } from '@testing-library/react'
import Cell from '../../../src/client/components/Cell'
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });
describe('Cell component', () => {

  const Wrapper = () => (
    <Cell />
  )

  it('Is exists', () => {
    const wrapper = shallow(<Wrapper />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Can mount', () => {
    const Wr = () => (
      <Cell
        type={1}
        isOtherUser={false}
      />
    )
    const w = render(<Wr />);
    expect(w).not.toBeNull()
  })
});