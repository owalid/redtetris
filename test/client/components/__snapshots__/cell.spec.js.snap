// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell component Is exists 1`] = `<Cell />`;
